src/aragonjs-wrapper.js

```diff
@@ -72,24 +210,124 @@ const initWrapper = async (
     throw err
   }
 
-  const { apps, forwarders, transactions } = wrapper
+  const subscriptions = subscribe(
+    wrapper,
+    {
+      onApps,
+      onForwarders,
+      onTransaction,
+    },
+    {
+      ipfsConf,
+    }
+  )
 
-  const subscriptions = {
-    apps: apps.subscribe(apps => {
-      const frontendApps = prepareFrontendApps(apps)
-      onApps(frontendApps, apps)
-    }),
-    forwarders: forwarders.subscribe(onForwarders),
-    transactions: transactions.subscribe(onTransaction),
+  wrapper.connectAppIFrame = (iframeElt, proxyAddress) => {
+    const provider = new providers.WindowMessage(iframeElt.contentWindow)
+    const result = wrapper.runApp(provider, proxyAddress)
+    if (subscriptions.connectedApp) {
+      subscriptions.connectedApp.unsubscribe()
+    }
+    subscriptions.connectedApp = result.shutdown
+    return result
   }
 
   wrapper.cancel = () => {
     Object.values(subscriptions).forEach(subscription => {
-      subscription.unsubscribe()
+      if (subscription) {
+        subscription.unsubscribe()
+      }
     })
   }
 
   return wrapper
 }
 
+const templateParamFilters = {
+  democracy: (
+    // supportNeeded: Number between 0 (0%) and 1 (100%).
+    // minAcceptanceQuorum: Number between 0 (0%) and 1 (100%).
+    // voteDuration: Duration in seconds.
+    { supportNeeded, minAcceptanceQuorum, voteDuration },
+    account
+  ) => {
+    const tokenBase = Math.pow(10, 18)
+    const percentageBase = Math.pow(10, 18)
+    const holders = [{ address: account, balance: 1 }]
+
+    const [accounts, stakes] = holders.reduce(
+      ([accounts, stakes], holder) => [
+        [...accounts, holder.address],
+        [...stakes, holder.balance * tokenBase],
+      ],
+      [[], []]
+    )
+
+    return [
+      accounts,
+      stakes,
+      supportNeeded * percentageBase,
+      minAcceptanceQuorum * percentageBase,
+      voteDuration,
+    ]
+  },
+
+  multisig: (
+    // signers: Accounts corresponding to the signers.
+    // neededSignatures: Minimum number of signatures needed.
+    { signers, neededSignatures },
+    account
+  ) => {
+    if (!signers || signers.length === 0) {
+      throw new Error('signers should contain at least one account:', signers)
+    }
+
+    if (neededSignatures < 1 || neededSignatures > signers.length) {
+      throw new Error(
+        'neededSignatures must be between 1 the total number of signers',
+        neededSignatures
+      )
+    }
+
+    return [signers, neededSignatures]
+  },
+}
+
+export const initDaoBuilder = (
+  provider,
+  registryAddress,
+  ipfsConf = ipfsDefaultConf
+) => {
+  // DEV only
+  // provider = new Web3.providers.WebsocketProvider('ws://localhost:8546')
```

Comment:
We could probably put this in the environment, as `providers.daoBuilder` (and set it to be the same as the wallet provider for non-dev)?